docs/source/cuda/cuda_array_interface.rst

```diff
@@ -69,13 +69,213 @@ The following are optional entries:
+   version of the CAI. A future revision to the interface specification may
+   consider the use of multiple host threads once the current version and its
+   implementations have matured and become well-understood.
+
+
+Interface Requirements
+~~~~~~~~~~~~~~~~~~~~~~
+
+The ``sync`` and ``stream`` items of the interface indicate whether and how
+synchronization is expected so that producers and consumers that operate
+asynchronously can avoid hazards when exchanging data through the CUDA Array
+Interface.
+
+From the Consumer's perspective:
+
+* ``stream`` indicates the stream on which the Producer may have in-progress
+  operations on the data, and which the Consumer may be expected to either:
+
+  - synchronize on before accessing the data, or,
+  - enqueue operations in when accessing the data.
+
+  Whether one of these requirements applies depends partly on the ``sync``
+  parameter, and partly on the User's choice.
+
+* ``sync`` indicates the requirement for synchronization on the given
+  ``stream``.
+
+  - When ``sync`` is ``False``, or the User has set the Consumer to ignore CAI
+    synchronization semantics, the Consumer may assume that operating
+    on the data is immediately permitted, with no further synchronization.
+  - When ``sync`` is ``True``, and the User has not overridden CAI
+    synchronization semantics, then the Consumer must either:
+
+    - Synchronize on the provided stream prior to accessing the data. If after
+      synchronization the Consumer operates on the data in another stream, it
+      must ensure that no computation can take place in the provided stream
+      until its operations in its own choice of stream have taken place. This
+      could be achieved by either:
+
+      - Placing a wait on an event in the provided stream that occurs once all
+        of the Consumer's operations on the data are completed, or
+      - Avoiding returning control to the user code until after its operations
+        on its own stream have completed.
+
+    - Only enqueuing operations on the data in the provided stream. The Consumer
+      may return control to the User code immediately after enqueueing its work,
+      as the work will all be serialized on the exported array's stream, ensuring
+      correctness even if the User code were to induce the Producer to
+      subsequently start enqueueing more work on the same stream.
+
+When exporting an array through the CAI, Producers must ensure that:
+
+* There is work on the data enqueued on at most one stream.
+* If there is work on the data enqueued on a stream, then ``sync`` should be
+  ``True``, and ``stream`` is the stream on which pending operations on the data
+  are enqueued. There must not be any work enqueued on the data in any other
+  streams.
+* If there is no work enqueued on the data, then ``sync`` should be ``False``,
+  and ``stream`` may be either ``None``, or a stream on which it is suggested
+  that work on the data is enqueued.
```

Comment:
> on which it is suggested that work on the data is enqueued.

What is a use case for a suggested stream?

Reply:
Given that if `sync=False` it's safe to use the data on any stream, and if `sync=True` then you need to either synchronize the given stream or enqueue onto the given stream (unless we add an awaitable event), it makes a lot more sense to me that we just remove the `sync` parameter

Reply:
I can't think of a use case I might have been imagining - this wording is now removed.